MAC_SETUP.md

```diff
@@ -97,35 +101,76 @@ test their code.
 
-    ```console
-    cd ~/src/python-docs-samples
-    ```
+## Python virtual environments
+Using [Virtual Environments](https://docs.python.org/3/library/venv.html) prevents inadvertent modifications to your global python install. Once created and sourced, calls to `python` will uses this virtual environment, not a global python install. Each virtual environment can have its own set of packages that can be different from others.
 
-1.  Create a virtualenv using `pyenv virtualenv`.
 
-    ```console
-    pyenv virtualenv 3.7.6 python-docs-samples
-    ```
+### Using Python 3+ venv
+Python has builtin support for creating virtual environments, accessible by running the `venv` module.
 
-    This creates a virtualenv folder within `~/.pyenv/versions/`.
+```
+cd python-docs-samples
+python -m venv [venv-name]
+source [venv-name]/bin/activate
+```
 
-1.  Set the local Python version(s) with `pyenv local`
+Typically you will name the venv `venv`, or `venv38` for a python 3.8 venv.
 
-    ```console
-    # pyenv local [name of virtualenv] [list of python versions to use]
-    pyenv local python-docs-samples 3.6.10 3.7.6 3.5.9 2.7.17
-    ```
 
-1.  Now, when you `cd` into the source directory or a subdirectory within it,
-    pyenv will make your virtualenv the default Python. Since you specified
-    more than one version, it will also add binaries like `python36` and
-    `python27` to your PATH, which nox uses when picking Python interpreters.
+### Using pyenv-virtualenv
+You can also use an extension for pyenv that will assist in managing virtual environments. This allows you to use `pyenv local` to automatically use the created virtual environment. You can install this by running `$ brew install pyenv-virtualenv`
+
+1.  Change to the desired source directory.	
+
+    ```console	
+    cd ~/src/python-docs-samples	
+    ```	
+
+1.  Create a virtualenv for python 3.8.1 using `pyenv virtualenv`.	
 
-1.  Add `.python-version` to your [global gitignore
-    file](https://help.github.com/articles/ignoring-files/#create-a-global-gitignore),
+    ```console	
+    pyenv virtualenv 3.8.1 python-docs-samples	
+    ```	
+
+    This creates a virtualenv folder within `~/.pyenv/versions/`.	
+
+1.  Set the local Python version(s) with `pyenv local`	
+
+    ```console	
+    # pyenv local [name of virtualenv] [list of python versions to use]	
+    pyenv local python-docs-samples 3.8.1 3.7.6 3.6.10 3.5.9 2.7.17	
+    ```	
+
+1.  Now, when you `cd` into the source directory or a subdirectory within it,	
+    pyenv will make your virtualenv the default Python. Since you specified	
+    more than one version, it will also add binaries like `python36` and	
+    `python27` to your PATH, which nox uses when picking Python interpreters.	
+
+1.  Add `.python-version` to your [global gitignore	
+    file](https://help.github.com/articles/ignoring-files/#create-a-global-gitignore),	
```

Comment:
Nit: this rendered link looks a bit weird separated by a newline, although that could be my markdown rendering tool. If the line is too long, you could put the link at the bottom and reference it here with a nickname (e.g. the last line of [this markdown](https://raw.githubusercontent.com/googleapis/nodejs-translate/master/CONTRIBUTING.md)).

Reply:
Thanks for finding that. It looked good in my previewer, but I combined links to one line just to be safe.